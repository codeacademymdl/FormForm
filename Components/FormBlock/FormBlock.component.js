import React, { Component } from 'react';
import {
  StyleSheet, Text, View,
} from 'react-native';

import moment from 'moment';
import styles from './FormBlock.style';

// moment("2010-10-20 4:30 +0000", "YYYY-MM-DD HH:mm Z")


export default class FormBlock extends Component {
  // date = moment(this.props.date, 'YYYY-MM-DD HH:mm');

  // formatDate = this.date._d;


  render() {
    // console.log('this date', this.date._d);
    return (
      <View style={styles.main}>
        <Text style={styles.content}>{this.props.title}</Text>
        <View style={styles.bottomView}>
          <Text style={styles.bottomText}>
            Created On:
            {this.props.date}
          </Text>
        </View>
      </View>

    );
  }
}
