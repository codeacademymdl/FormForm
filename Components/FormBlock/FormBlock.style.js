import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  main: {
    flex: 1,
    backgroundColor: 'white',
    justifyContent: 'center',
    width: '70%',
    height: 200,
    marginTop: 10,
    marginBottom: 10,
  },
  content: {
    color: 'black',
    fontSize: 30,
    marginLeft: 10,
  },
  bottomView: {
    marginTop: '30%',
    borderTopWidth: 2,
    borderTopColor: 'black',
    justifyContent: 'flex-end',
    alignItems: 'flex-end',
    marginRight: 10,

  },
  bottomText: {
    fontSize: 15,
  },
});
