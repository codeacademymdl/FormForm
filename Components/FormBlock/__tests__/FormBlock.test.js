import React from 'react';

import renderer from 'react-test-renderer';
import FormBlock from '../FormBlock.component';


describe('FillForm', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<FormBlock title="ThisTitle" date="10thDecember 2019" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
