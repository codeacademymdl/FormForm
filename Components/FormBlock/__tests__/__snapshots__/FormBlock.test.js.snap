// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FillForm renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "height": 200,
      "justifyContent": "center",
      "marginBottom": 10,
      "marginTop": 30,
      "width": "70%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 30,
        "marginLeft": 10,
      }
    }
  >
    ThisTitle
  </Text>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "borderTopColor": "black",
        "borderTopWidth": 2,
        "justifyContent": "flex-end",
        "marginRight": 10,
        "marginTop": "30%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 15,
        }
      }
    >
      Created On:
      10thDecember 2019
    </Text>
  </View>
</View>
`;
